plugin/hotswap-agent-spring-plugin/src/main/java/org/hotswap/agent/plugin/spring/core/BeanFactoryProcessor.java

```diff
@@ -0,0 +1,131 @@
+            if (AnnotatedBeanDefinitionUtils.getFactoryMethodMetadata(annotatedBeanDefinition) != null) {
+                return false;
+            }
+        }
+        if (resolveBeanClassMethod != null) {
+            resolveBeanClassMethod.setAccessible(true);
+            Class<?> beanClass = null;
+            BeanDefinition rBeanDefinition = beanFactory.getMergedBeanDefinition(beanName);
+            try {
+                if (rBeanDefinition != null) {
+                    beanClass = (Class<?>) resolveBeanClassMethod.invoke(beanFactory, rBeanDefinition, beanName, new Class[]{});
+                }
+            } catch (IllegalAccessException e) {
+                LOGGER.warning("resolveBeanClass error", e);
+                return false;
+            } catch (InvocationTargetException e) {
+                LOGGER.warning("resolveBeanClass error", e);
+                return false;
+            } catch (Exception e) {
+                LOGGER.warning("resolveBeanClass error", e);
+                return false;
+            }
+            Method method = ReflectionUtils.findMethod(beanFactory.getClass(), "determineConstructorsFromBeanPostProcessors", Class.class, String.class);
+            if (method != null) {
+                method.setAccessible(true);
+                try {
+                    Constructor<?>[] constructors = (Constructor<?>[]) method.invoke(beanFactory, beanClass, beanName);
+                    if (constructors != null && constructors.length > 0) {
+                        return predicate.test(constructors);
+                    }
+                } catch (Exception e) {
+                    LOGGER.error("determineConstructorsFromBeanPostProcessors error", e);
+                }
+            }
+        }
+        return false;
+    }
+
+    private static void resetEmbeddedValueResolvers(DefaultListableBeanFactory beanFactory, String beanName) {
+        Object target = beanFactory.getSingleton(beanName);
+        if (target != null && target instanceof PlaceholderConfigurerSupport && target instanceof IPlaceholderConfigurerSupport) {
+            IPlaceholderConfigurerSupport placeholderConfigurerSupport = (IPlaceholderConfigurerSupport) target;
+            Field field = ReflectionUtils.findField(beanFactory.getClass(), "embeddedValueResolvers");
+            if (field != null) {
+                ReflectionUtils.makeAccessible(field);
+                List<StringValueResolver> embeddedValueResolvers = (List<StringValueResolver>) ReflectionUtils.getField(field, beanFactory);
+                embeddedValueResolvers.removeAll(placeholderConfigurerSupport.valueResolvers());
+            }
+        }
+    }
+
+    public static boolean isAllowBeanDefinitionOverriding(DefaultListableBeanFactory beanFactory) {
+        Object target = ReflectionHelper.getNoException(beanFactory, beanFactory.getClass(), "allowBeanDefinitionOverriding");
```

Comment:
I noticed that you use reflection API to get 'allowBeanDefinitionOverriding' but directly use setter to set 'allowBeanDefinitionOverriding'. If this is necessary (e.g, to deal with the different spring releases), would you mind to add more comments above the methods?

Reply:
org.springframework.beans.factory.support.DefaultListableBeanFactory.isAllowBeanDefinitionOverriding is introduced in spring 4.1.2